services/packages/packages.go

```diff
@@ -340,7 +340,11 @@ func DeletePackageVersionAndReferences(ctx context.Context, pv *packages_model.P
 		}
 	}
 
-	return packages_model.DeleteVersionByID(ctx, pv.ID)
+	if err := packages_model.DeleteVersionByID(ctx, pv.ID); err != nil {
+		return err
+	}
+
+	return deleteUnreferencedPackages(ctx)
```

Comment:
Isn't this is essentially doing a lot of the work that is in cleanup on every delete? i.e. it's tantamount to calling CleanUp for every delete?

Cant we use the information in pv to limit our unreferenced check to that particular package?

Reply:
It's not really much work. In the best case it's 1 sql query, average case is 1+2 queries and the worst case is 1+n*2 queries.

Anyway I just changed the definition of `HasOwnerPackages`, so no additional work to do.